core/src/BibleEngine.class.ts

```diff
@@ -403,6 +403,9 @@ export class BibleEngine {
         const entityManager = await this.pEntityManager;
         let firstPhraseId: number | undefined, lastPhraseId: number | undefined;
         for (const content of contents) {
+            if (!content) {
```

Comment:
So, even though we don't allow nullable types, types can still be undefined. I couldn't find the root cause of undefined phrases/sections/paragraphs (whatever the content might be), so I just filter them out here. Maybe that's bad, but its a nice safeguard

Reply:
actually typescript checks for undefined as well. as far as TypeScript is concerned `content` can't be anything but an `IBibleContentForInput` object. if i'm not overlooking something, this can only happen if you use `any` somewhere, i.e. if you opt out of typing somewhere in your importer (or if you force-cast an object)..

i'm not against safeguards but this indicates that there is a bug somewhere in the code, so just silencing it - even if you know what you are doing - will just produce other problems down the line. If we really need a safeguard at this place i would suggest to throw an error instead.

so if you really feel like you have a good reason that there are falsy items in `contents`, would it be possible to instead put the filter in the importer (that is calling the method)? Otherwise we introduce a potential bug for every other code calling the method. In those cases i would also put a comment that is stating the reasons for doing so. if it's an undiscovered bug that we choose to live with temporarily i would additionally add a `TODO: ` that includes the information about the bug that is available already.

Reply:
You're right, I need to address the underlying problem. I'll see if I can fix it

Reply:
fixed!